fix: initOnceGrpc function bugs

samlaf opened the pull request:
Closes EGDA-1742.

These functions had a bug where the error was not stored, and hence would not be re-returned on future calls.

Furthermore, realized the function itself is completely unnecessary because creating a new grpc client doesn't actually make any IO calls. Those are only done the first time an rpc is submitted. So this logic was not needed at all. Removed it.

## Why are these changes needed?

<!-- Please give a short summary of the change and the problem this solves. -->

## Checks

- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- [ ] I've checked the new test coverage and the coverage percentage didn't drop.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

bxue-l2 reviewed (approved):
lgtm, how do you discover those bugs

samlaf commented:
> Ok, I've updated #1815 (fixed merge conflicts, light cleanup). Should be ready to merge with an approval.

merged master into this PR and fixed conflicts with your changes in v2/disperser_client.go, can you take a look and re-approve if it looks good?
